modules/tensor_mechanics/examples/wave_propagation/tests

```diff
@@ -0,0 +1,19 @@
+[Tests]
+  issues = '#21364'
+  design = '1d_elastic_waves.md'
+
+  [test_real]
+    type = 'CSVDiff'
+    input = '1D_elastic_wave_propagation.i'
+    csvdiff = '1D_elastic_wave_propagation_out_midpt_real_0001.csv'
+    valgrind = NONE
+    requirement = "Validating the frequency domain solution against the analytical solution"
```

Comment:
The system shall validate the frequency domain solution of the real part against the analytical solution.